pandas/core/indexes/multi.py

```diff
@@ -1840,15 +1849,19 @@ def is_lexsorted(self) -> bool:
         ...                            ['bb', 'aa', 'aa', 'bb']]).is_lexsorted()
         False
         """
-        return self.lexsort_depth == self.nlevels
+        return self._lexsort_depth == self.nlevels
 
-    @cache_readonly
+    @property
     def lexsort_depth(self):
-        if self.sortorder is not None:
-            return self.sortorder
-
-        return self._lexsort_depth()
+        warnings.warn(
+            "MultiIndex.is_lexsorted is deprecated as a public function, "
```

Comment:
@MarcoGorelli should the message here refer to lexsort_depth instead of is_lexsorted?

this warning isn't being caught in a bunch of tests; not sure why that isn't caught by the npdev build

Reply:
yes, looks like a typo, thanks for catching - we probably want to always use `match=` when catching warning to help prevent this, i'll fix this now

> this warning isn't being caught in a bunch of tests; not sure why that isn't caught by the npdev build

not sure what you mean here sorry